jetty-core/jetty-http/src/test/java/org/eclipse/jetty/http/HttpCookieTest.java

```diff
@@ -105,4 +110,123 @@ public void testParseInvalidAttributes(String name, String value, Class<? extend
+        // unexpected timezone (the timezone is ignored per spec)
+        args.add(Arguments.of("Sun, 06 Nov 1994 08:49:37 PST", "1994/11/06 08:49:37 GMT"));
+        // long weekday (weekday is ignored per spec)
+        args.add(Arguments.of("Wednesday, 09 Jun 2021 10:18:14 GMT", "2021/06/09 10:18:14 GMT"));
+
+        return args.stream();
+    }
+
+    @ParameterizedTest
+    @MethodSource("cookieDatesValid")
+    public void testParseCookieDate(String input, String expected)
+    {
+        Instant actual = HttpCookie.parseCookieDate(input);
+        DateTimeFormatterBuilder formatter = new DateTimeFormatterBuilder();
+        String actualStr = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss O").format(actual.atZone(ZoneId.of("GMT")));
+        assertEquals(expected, actualStr);
+    }
+
+    public static Stream<Arguments> cookieDatesInvalid()
+    {
+        List<Arguments> args = new ArrayList<>();
+
+        // Preferred RFC 1123 syntax
+        // - invalid Year
+        args.add(Arguments.of("Sun, 06 Nov 65535 08:49:37 GMT", "Missing [year]"));
+        args.add(Arguments.of("Wed, 09 Jun -123 10:18:14 GMT", "Missing [year]"));
+        // - no year
+        args.add(Arguments.of("Wed, 09 Jun 10:18:14 GMT", "Missing [year]"));
+
+        // - invalid day
+        args.add(Arguments.of("Tue, 00 Apr 2024 17:28:27 GMT", "Invalid [day]"));
+        // - no day
+        args.add(Arguments.of("Thu, Nov 2017 13:37:22 GMT", "Missing [day]"));
+        // - single digit hour
+        args.add(Arguments.of("Wed, 09 Jun 2021 2:18:14 GMT", "Missing [time]"));
+
+        // - no time
+        args.add(Arguments.of("Thu, 01 Jan 1970 GMT", "Missing [time]"));
+
+        // Obsolete RFC 850 syntax
+        // - invalid year
+        args.add(Arguments.of("Sun, 03-May-65535 18:54:31 GMT", "Missing [year]"));
+        // - no year
+        args.add(Arguments.of("Tue, 16-Apr- 17:28:27 GMT", "Missing [year]"));
+        // - invalid day
+        args.add(Arguments.of("Sunday, 00-Nov-94 08:49:37 GMT", "Invalid [day]"));
+        // - single digit hour
+        args.add(Arguments.of("Sunday, 01-Nov-94 8:49:37 GMT", "Missing [time]"));
+
+        // - no day (the '94' is parsed as day as its the first 2 digit number in the string)
+        args.add(Arguments.of("Sun, Nov-94 08:49:37 GMT", "Missing [year]"));
+
+        // - no time
+        args.add(Arguments.of("Mon, 05-May-2014 GMT", "Missing [time]"));
+        // - bad time (no seconds) - will not find time
+        args.add(Arguments.of("Thu, 02-May-2013 13:14 GMT", "Missing [time]"));
+        // - bad time (am/pm) - will not find time, and will not understand the AM/PM
+        args.add(Arguments.of("Thu, 02-May-2013 13:14 PM GMT", "Missing [time]"));
+
+        // Obsolete ANSI C's asctime() format
+        // - invalid year
+        args.add(Arguments.of("Sun Nov  6 08:49:37 65535", "Missing [year]"));
+
+        return args.stream();
+    }
```

Comment:
How about adding:

* long month
* negative day, hour, minute, second
* More than 1 space between day and month and year
* What happens if datetime is earlier than EPOCH?